npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/registry_finder.rb

```diff
@@ -121,8 +121,9 @@ def locked_registry
             known_registries.
             find { |h| h["registry"].include?(lockfile_registry) }&.
```

Comment:
Should we try to find known registries using only the host first?

Reply:
My intuition is that this `lockfile_registry` variable should be holding the proper registry url already (just host, or host plus some path depending on the type of private registry). The fact that's holding a full path to a dependency in this case I believe it may be a bug in the file parser, specific to GitHub private registries. But that's just me guessing...